src/app/dataset-view/dataset.component.ts

```diff
@@ -210,49 +225,50 @@ export class DatasetComponent
         });
     }
 
-    public onClickNode(idDataset: string): void {
+    public onClickNode(node: any): void {
+        this.selectedDatasetName = node.label;
```

Comment:
I don't think `selectedDatasetName` should be a property of the component. It's not used anywhere in templates. You just need to pass it as an argument to `onSelectDataset`.

Reply:
Fixed